### PR TITLE
Update grafana-plugin-go-sdk to v0.185.0 and otel to v1.19.0

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**

Upgrades grafana-plugin-go-sdk based on https://github.com/grafana/grafana-plugin-sdk-go/pull/773 and otel libraries. Alternative to #76677.

Note: Need to use proper SDK version when that has been released.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -503,7 +501,6 @@ replace xorm.io/xorm => github.com/grafana/xorm v0.8.3-0.20230627081928-d04aa38a
 // This is required in order to get notification delivery errors from the receivers API.
 replace github.com/prometheus/alertmanager => github.com/grafana/prometheus-alertmanager v0.25.1-0.20230918083811-3513be6760b7
 
-// grpc v1.46.0 removed "WithBalancerName()" API, still in use by weaveworks/commons.
-replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
-
 exclude github.com/mattn/go-sqlite3 v2.0.3+incompatible
+
+replace github.com/grafana/grafana-plugin-sdk-go => github.com/grafana/grafana-plugin-sdk-go v0.184.1-0.20231017153757-4a53ea278945
```
Comment (andresmgot) [inline go.mod:506]: just a note to not forget to release a new version of the sdk and replace it in line 68 instead

**Review by andresmgot — APPROVED** [review]
works for me!

File: go.mod (hunk 1 above)
Comment (marefr) [reply go.mod]: Done

**Comment by VeenaThimmegowda** [comment]
Is this merge will also have fix for CVE-2023-44487??

**Pull request merged**